guides/source/asset_pipeline.md

```diff
@@ -3,561 +3,545 @@
 
-You can call this command on the server during deployment to create compiled
-versions of your assets directly on the server. See the next section for
-information on compiling locally.
+In production, precompilation is typically run during deployment to ensure that
+the latest versions of the assets are served. Propshaft was explicitly not
+designed to provide full transpiler capabilities. However, it does offer an
+input -> output compiler setup that by default is used to translate `url(asset)`
+function calls in CSS to `url(digested-asset)` instead and source mapping
+comments likewise.
 
-The command is:
+To manually run precompilation you can use the following command:
 
 ```bash
 $ RAILS_ENV=production rails assets:precompile
 ```
 
-This links the folder specified in `config.assets.prefix` to `shared/assets`.
-If you already use this shared folder you'll need to write your own deployment
-command.
-
-It is important that this folder is shared between deployments so that remotely
-cached pages referencing the old compiled assets still work for the life of
-the cached page.
+After doing this, all assets in the load path will be copied (or compiled when
+using [advanced asset management](#advanced-asset-management)) in the
+precompilation step and stamped with a digest hash.
 
-NOTE. Always specify an expected compiled filename that ends with `.js` or `.css`.
+Additionally, you can set ENV["SECRET_KEY_BASE_DUMMY"] to trigger the use of a
+randomly generated secret_key_base that’s stored in a temporary file. This is
+useful when precompiling assets for production as part of a build step that
+otherwise does not need access to the production secrets.
 
-The command also generates a `.sprockets-manifest-randomhex.json` (where `randomhex` is
-a 16-byte random hex string) that contains a list with all your assets and their respective
-fingerprints. This is used by the Rails helper methods to avoid handing the
-mapping requests back to Sprockets. A typical manifest file looks like this:
-
-```json
-{"files":{"application-<fingerprint>.js":{"logical_path":"application.js","mtime":"2016-12-23T20:12:03-05:00","size":412383,
-"digest":"<fingerprint>","integrity":"sha256-<random-string>"}},
-"assets":{"application.js":"application-<fingerprint>.js"}}
+```bash
+$ RAILS_ENV=production SECRET_KEY_BASE_DUMMY=1 rails assets:precompile
 ```
 
-In your application, there will be more files and assets listed in the manifest,
-`<fingerprint>` and `<random-string>` will also be generated.
-
-The default location for the manifest is the root of the location specified in
-`config.assets.prefix` ('/assets' by default).
+By default, assets are served from the `/assets` directory.
 
-NOTE: If there are missing precompiled files in production you will get a
-`Sprockets::Helpers::RailsHelper::AssetPaths::AssetNotPrecompiledError`
-exception indicating the name of the missing file(s).
+WARNING: Do not run the precompile command in development mode. Running it in
```

Comment:
I think precompiling in development can be useful for debugging. My suggestion would be to soften the warning.


```suggestion
WARNING: Running the precompile command in
```

Reply:
Out of scope for this guide, but it may also be valuable for Rails to print a warning with these explanations when running `assets:precompile` in unexpected environments.